feat(flags): Use Contour request ID from x-request-id header

haacked opened the pull request:
## Problem

The `/flags` service generates its own unique request id on every request. However, the request to `/flags` originates in Contour which passes flags a request id via the `x-request-id` header.

## Changes

Use the existing request ID from Contour's `x-request-id` header instead of generating a new random UUID for better request tracing consistency.

Fixes #37770

## How did you test this code?

Unit tests
Manually

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

haacked committed:
feat(flags): Use Contour request ID from x-request-id header

Use the existing request ID from Contour's `x-request-id` header instead of generating a new random UUID for better request tracing consistency.

Fixes #37770

dmarticus reviewed (approved):
good as always, one comment about the comment

haacked committed:
Add comment on why this is useful

haacked committed:
Use all-caps as it's more conventional

HTTP header field names are case insensitive, but it looks weird to have it lower case.
